src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -0,0 +1,93 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  ids: string;
+}
+
+class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_RESTORE;
+  }
+
+  public get description(): string {
+    return 'Restores given items from the site recycle bin';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Restoring items from recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
+
+    const ids: string[] = this.splitIdsList(args.options.ids);
+    const idsChunks: string[][] = [];
+
+    while (ids.length) {
+      idsChunks.push(ids.splice(0, 20));
+    }
+
+    idsChunks.forEach(async (idsChunk: string[], index: number) => {
+      const requestOptions: any = {
+        url: requestUrl,
+        headers: {
+          'accept': 'application/json;odata=nometadata'
+        },
```

Comment:
Since we're issuing a POST request, we should specify the `content-type` header.

Reply:
done